ingest/expression_files/mtx.py

```diff
@@ -64,19 +64,16 @@ def __init__(self, mtx_path: str, study_file_id: str, study_id: str, **kwargs):
             self.mtx_local_path, 2).split()
 
     def execute_ingest(self):
-        # import pdb
-        # pdb.set_trace()
         self.extract_feature_barcode_matrices()
-        # import pdb
-        # pdb.set_trace()
-        yield from self.transform()
-        self.close()
+        for gene_docs, data_array_documents in self.transform():
+            self.load(gene_docs, data_array_documents)
+        return 0
 
     def extract_mtx(self, value):
         """
         Zgreps by gene index from mtx file to enhance performance/scale
```

Comment:
Reflecting your update in line 79:

```suggestion
        greps by gene index from mtx file to enhance performance/scale
```

Out of curiosity, what makes `grep` suitable but not `zgrep`?  (Do we know why `grep` performs better?)

Reply:
zgrep is used for zipped files and was needed to ingest the largest matrix, a zipped file, in PAPI. When trying to fulfill [SCP-2577](https://broadworkbench.atlassian.net/browse/SCP-2577) is wasn't as performant as expected. The files we use for testing aren't zipped which is why I needed to revert zgrep to grep. I believe this method will become less relevant due to [SCP-2661](https://broadworkbench.atlassian.net/browse/SCP-2611#) and the fact that it wasn't performant.

Reply:
cool, don't forget to update the comment

Reply:
cool, don't forget to update the comment

Reply:
don't forget to update the comment